### ISSUE TITLE
Not getting the call back using the setImage method.

### ISSUE THREAD

**Issue by ramprasadios** [onevcat/Kingfisher#1333]
Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

Not getting call back in the setImage method, I have a requirement to display the thumbnail image in the collectionView, The thumbnail images are very large in size, so i have downsampled the image using the processors of Kingfisher processor APIs. First time i am able to retrieve the images and display in the collection View cells. But when i come back to the same screen, i am not getting the images, and the setImage callback is not being called and i am not sure weather the images are failed or downloaded.

I am using the below code to cache and display the image to the collection view grid cells:

                    if let url = URL(string: imageUrl ?? "") {
                            let processor = DownsamplingImageProcessor(size: cell.imageView.bounds.size)
                            cell.imageView.kf.indicatorType = .activity
                            cell.imageView.contentMode = .scaleAspectFit
                            
                            cell.imageView.kf.setImage(
                                with: url,
                                placeholder: UiElements.placeholderImage,
                                options: [
                                    .processor(processor),
                                    .scaleFactor(UIScreen.main.scale),
                                    .transition(.fade(1)),
                                    .fromMemoryCacheOrRefresh
                                ])
                            {
                                result in
                                switch result {
                                case .success(let value):
                                    cell.imageView.contentMode = .scaleToFill
                                    print("Task done for: \(value.source.url?.absoluteString ?? "")")
                                case .failure(let error):
                                    cell.imageView.image = UiElements.placeholderImage
                                    print("Job failed: \(error.localizedDescription)")
                                }
                            }
                        }

**Comment by onevcat** [comment]
@ramprasadios 

Have you ever tried once `.cacheOriginalImage` option in your code or not? I did find an issue when firstly `. cacheOriginalImage` set and then it is removed while `.fromMemoryCacheOrRefresh ` is still there at the same time. Let me know if it is your case.

Before we can get a conclusion and stable reproducible steps, I think you can remove the `.fromMemoryCacheOrRefresh` option for a workaround. Or you can try to remove all the cache once like:

```swift
KingfisherManager.shared.cache.clearMemoryCache()
KingfisherManager.shared.cache.clearDiskCache()
```

**Comment by ramprasadios** [comment]
Since i am working on a scenario, where in there will be more than 200+ images to be displayed in the collectionView. if i use .` cacheOriginalImage` option, i will end-up getting a crash due to high consumption of memory by iOS. so i have not used that option.

**Comment by onevcat** [comment]
I had a fix for the issue I mentioned above in #1351 

However, I am not sure whether it is related to your issue or not (probably not...)

Can you try the [`fix/fromMemoryCacheOrRefresh-skip`](https://github.com/onevcat/Kingfisher/tree/fix/fromMemoryCacheOrRefresh-skip) branch and let me know the result? (maybe you will get an assertion, etc.)

**Comment by onevcat** [comment]
Merged to master.
